tests/framework/clients/corral/corral.go

```diff
@@ -30,24 +36,49 @@ func SetupCorralConfig(configVars map[string]string) error {
 	return nil
 }
 
+// SetCustomRepo sets a custom repo for corral to use. It takes a string as a parameter which is the repo you want to use
+func SetCustomRepo(repo string) error {
+	msg, err := exec.Command("git", "clone", repo, "corral-packages").CombinedOutput()
+	if err != nil {
+		return errors.Wrap(err, "Failed to git clone remote repo: "+string(msg))
+	}
+	makemsg, err := exec.Command("make", "init", "-C", "corral-packages").CombinedOutput()
+	if err != nil {
+		return errors.Wrap(err, "Failed to git clone remote repo: "+string(makemsg))
+	}
+	makebuildmsg, err := exec.Command("make", "build", "-C", "corral-packages").CombinedOutput()
+	if err != nil {
+		return errors.Wrap(err, "Failed to git clone remote repo: "+string(makebuildmsg))
+	}
+	logrus.Infof("Successfully set custom repo: %s", repo)
+	return nil
+}
+
 // CreateCorral creates a corral taking the corral name, the package path, and a debug set so if someone wants to view the
 // corral create log
-func CreateCorral(corralName, packagePath string, debug bool) (*rest.Config, error) {
-	if debug {
-		msg, err := exec.Command("corral", "create", corralName, packagePath, debugFlag).CombinedOutput()
-		if err != nil {
-			return nil, errors.Wrap(err, "CreateCorral: "+string(msg))
-		}
-
-		myString := string(msg[:])
-		logrus.Infof(myString)
-	} else {
-		msg, err := exec.Command("corral", "create", corralName, packagePath).CombinedOutput()
-		if err != nil {
-			return nil, errors.Wrap(err, "CreateCorral: "+string(msg))
-		}
+func CreateCorral(ts *session.Session, corralName, packageName string, debug bool, cleanup bool) (*rest.Config, error) {
+	if cleanup {
+		ts.RegisterCleanupFunc(func() error {
+			return DeleteCorral(corralName)
+		})
 	}
 
+	args := []string{"create"}
+	if !cleanup {
+		args = append(args, skipCleanupFlag)
+	}
+	if debug {
+		args = append(args, debugFlag)
+	}
+	args = append(args, corralName, packageName)
+	logrus.Infof("Creating corral with the following parameters: %", args)
+	// complicated, but running the command in a way that allows us to
+	// capture the output and error(s) and print it to the console
+	msg, err := exec.Command("corral", args...).CombinedOutput()
+	logrus.Infof("Corral create output: %s", string(msg))
+	if err != nil {
+		return nil, errors.Wrap(err, "Unable to create corral: "+string(msg))
```

Comment:
Did you figure out the issue with corral create not returning an error?

Reply:
[unfortunately not](https://github.com/rancher/rancher/pull/39752#issuecomment-1364263841).  However I believe the culprit here is Terraform, not necessarily corral.